frame/referenda/src/types.rs

```diff
@@ -405,6 +420,19 @@ impl Curve {
 				.checked_rounding_div(FixedI64::from(x) + *x_offset, Low)
 				.map(|yp| (yp + *y_offset).into_clamped_perthing())
 				.unwrap_or_else(Perbill::one),
+			Self::ExponentialDecay { n_0, half_time, offset } => {
+				//n_0 * (2 ^ -(x / half_time)) + offset
+				if let Some(exponent) = <Perbill as Into<I64F64>>::into(x).checked_div(*half_time) {
+					if let Ok(exponential) = pow(I64F64::from(2), -exponent) {
+						if let Some(decay) = n_0.checked_mul(exponential) {
+							if let Some(full_formula) = offset.checked_add(decay) {
+								return full_formula.into()
+							}
+						}
+					}
```

Comment:
This pattern can be replaced with a combination of  `map` and `and_then`, which is much more rust style

Reply:
sweet